### DISCUSSION TITLE
Displaying GLB blob in AR

### DISCUSSION THREAD

**Discussion by tanaydimri** [google/model-viewer#3588] [Q&A]
Hello All,
I have a use case where I need to download a threejs scene on the user's mobile device and display it in AR. 

Now, according to [this discussion](https://github.com/google/model-viewer/discussions/3184). I could only use webXR and Quick Look with this approach. What I did not know at the time is that webXR needs to be activated on each device, based on the errors that I am seeing in the console. Is my assumption correct? If yes, is there a way to at least ask for user's confirmation to enable webXR on their device for this session because 95% of our users would not be familiar with or willing to update their chrome setting just to view our scenes in AR.

Any pointers would be helpful here.

![image](https://user-images.githubusercontent.com/57341011/179733862-1b26cc4e-d928-4064-a851-8b0fc1bee300.png)

**Comment by elalish** [comment]
No, this isn't a Chrome settings issue. The problem is you're trying to start an AR session automatically, which browser security won't allow you to do. You need "user activation", which means something like a click event. If you use our normal AR button, it will work just fine.

**Reply by tanaydimri** [reply]
Ohh that is one of the scenarios, yes. I will take care of that, thank you!!
So if I understand this correctly, it would only work if the native AR button of the model-viewer is clicked, correct? Me calling modelViewerElement.activateAR() wouldn't work?

**Reply by elalish** [reply]
That will also work, provided you call `activateAR()` inside of something like a click event listener. If Chrome sees you trying to start WebXR without an associated user action, it'll stop you. Don't let it be too asynchronous.

**Reply by tanaydimri** [reply]
Aha!!
Thank you @elalish . I will keep those pointers in mind while fixing this tomorrow

**Reply by tanaydimri** [reply]
Hi @elalish 
Just wanted to confirm. I took your advice and everything seems to be working as expected now. Thank you again!!
